### PR TITLE
Make reading saver

### LINKED ISSUE #1935: Client crash on invalid mod
**Describe the bug**
User has mod installed with a bad file path and the client crashes on start up.

**To Reproduce**
Hard to say but I guess put some : in the path to the icon. But fix should be able without a way to reproduce.

**Log or error message**
This is my log file:
```
java.lang.RuntimeException: Exception in Application init method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:895)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'fileOpeningHandler' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/os/FileOpeningHandler.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'replayService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/replay/ReplayService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:893)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:879)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:140)
	at com.faforever.client.FafClientApplication.init(FafClientApplication.java:97)
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:824)
	... 2 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'replayService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/replay/ReplayService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 21 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 35 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:408)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 49 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:595)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 72 more
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at com.faforever.client.mod.ModVersion.lambda$fromModInfo$1(ModVersion.java:110)
	at java.base/java.util.Optional.map(Optional.java:258)
	at com.faforever.client.mod.ModVersion.fromModInfo(ModVersion.java:110)
	at com.faforever.client.mod.ModService.extractModInfo(ModService.java:267)
	at com.faforever.client.mod.ModService.extractModInfo(ModService.java:261)
	at com.faforever.client.mod.ModService.addMod(ModService.java:414)
	at com.faforever.client.mod.ModService.loadInstalledMods(ModService.java:153)
	at com.faforever.client.mod.ModService.onModDirectoryReady(ModService.java:126)
	at com.faforever.client.mod.ModService.lambda$afterPropertiesSet$0(ModService.java:110)
	at com.faforever.client.mod.ModService.afterPropertiesSet(ModService.java:113)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1855)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1792)
	... 83 more
```
**Expected behavior**
The client moves the mod to the corrupted folder, or just does not import this very mod. Also logs the behaviour. I think there is a message and special behaviour for corrupted mods just that it does not apply here for some reason.

**OS**
Windows

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by 1-alex98**
Also reads mods in the Task instead.
Fixes #1935

### CODE REVIEW

**Comment by 1-alex98** [comment]
Gimme reviews

File: src/main/resources/i18n/messages.properties [hunk 1]
```diff
@@ -792,6 +793,7 @@ mod.manager.reload=Reload
 mod.manager.simModsTooltip=SIM mods are such mods that effect the game play of all players.
 mod.manager.uiModsTooltip=UI mods are mods that only effect how you see the game.
 modVault.modManager=Manage active mods
+modVault.loadingMaps=Loading mods
```
Comment (Sheikah45) [inline src/main/resources/i18n/messages.properties:796]: should be modVault.loadingMods

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/mod/ModService.java [hunk 2]
```diff
@@ -110,7 +111,14 @@ public void afterPropertiesSet() {
...
     };
-    modDirectoryChangedListener.invalidated(preferencesService.getPreferences().getForgedAlliance().modsDirectoryProperty());
+    taskService.submitTask(new CompletableTask<Void>(Priority.LOW) {
+      @Override
+      protected Void call() throws Exception {
+        updateTitle(i18n.get("modVault.loadingMaps"));
+        modDirectoryChangedListener.invalidated(preferencesService.getPreferences().getForgedAlliance().modsDirectoryProperty());
+        return null;
+      }
+    });
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/mod/ModService.java:121]: Why use a task to invalidate manually? Why is this different than the maps?

Comment (1-alex98) [reply src/main/java/com/faforever/client/mod/ModService.java:121]: Cause it should be done in the background and maps are also loaded with a task

File: src/main/resources/i18n/messages.properties (hunk 1 above)
Comment (Sheikah45) [inline src/main/resources/i18n/messages.properties]: Hmm it deleted my comment so making it again.

Should be modVault.loadingMods

**Comment by 1-alex98** [comment]
Wanna merge after CI passed

**Pull request merged**